app/src/main/java/com/android/sample/ui/activity/CreateActivity.kt

```diff
@@ -99,277 +133,323 @@ fun CreateActivityScreen(
+                            profileViewModel.userState.value?.name ?: "",
+                            profileViewModel.userState.value?.surname ?: "",
+                            0)
+                    try {
+                      calendar.set(
+                          parts[2].toInt(),
+                          parts[1].toInt() - 1, // Months are 0-based
+                          parts[0].toInt(),
+                          0,
+                          0,
                           0)
-                  try {
-                    calendar.set(
-                        parts[2].toInt(),
-                        parts[1].toInt() - 1, // Months are 0-based
-                        parts[0].toInt(),
-                        0,
-                        0,
-                        0)
-                    val activity =
-                        Activity(
-                            uid = listActivityViewModel.getNewUid(),
-                            title = title,
-                            description = description,
-                            date = Timestamp(calendar.time),
-                            startTime = startTime,
-                            duration = duration,
-                            price = price.toDouble(),
-                            placesLeft = attendees.size.toLong(),
-                            maxPlaces = placesMax.toLongOrNull() ?: 0,
-                            creator = creator,
-                            status = ActivityStatus.ACTIVE,
-                            location = location,
-                            images = carouselItems.map { it },
-                            participants = attendees,
-                            type = types.find { it.name == selectedOption } ?: types[0],
-                            comments = listOf())
-                    listActivityViewModel.addActivity(activity)
-                    profileViewModel.addActivity(creator, activity.uid)
-                    navigationActions.navigateTo(Screen.OVERVIEW)
-                  } catch (_: NumberFormatException) {
-                    println("There is an error")
+                      val activity =
+                          Activity(
+                              uid = listActivityViewModel.getNewUid(),
+                              title = title,
+                              description = description,
+                              date = Timestamp(calendar.time),
+                              startTime = startTime,
+                              duration = duration,
+                              price = price.toDouble(),
+                              placesLeft = attendees.size.toLong(),
+                              maxPlaces = placesMax.toLongOrNull() ?: 0,
+                              creator = creator,
+                              status = ActivityStatus.ACTIVE,
+                              location = location,
+                              images = items.map { bitmapToBase64(it) },
```

Comment:
the list of bitmaps "items" is converted to a list of strings to be stored in the firestore database